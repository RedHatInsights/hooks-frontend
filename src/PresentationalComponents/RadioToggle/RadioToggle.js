import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Radio, ListItem } from '@patternfly/react-core';
import { BulletlessList } from 'PresentationalComponents';

export const ALL = 'all';
export const SELECTED = 'selected-only';

export class RadioToggle extends Component {
    static propTypes = {
        children: PropTypes.node,
        subject: PropTypes.string.isRequired,
        scope: PropTypes.string.isRequired,
        initial: PropTypes.oneOf([ ALL, SELECTED ]).isRequired,
        selectable: PropTypes.bool.isRequired,
        onToggle: PropTypes.func
    }

    handleChange = (_, event) => {
        const { value } = event.currentTarget;
        this.props.onToggle && this.props.onToggle(value);
        this.setState({ value });
    }

    constructor(props) {
        super(props);
        this.state = { value: props.initial };
    }

    render() {
        const { children, scope, selectable, subject } = this.props;
        const group = `${ scope }-${ subject.replace(' ', '-') }-radio`;
        return (
            <React.Fragment>
                <BulletlessList>
                    <ListItem>
                        <Radio value={ ALL }
                            defaultChecked={ !selectable || this.state.value === ALL }
                            onChange={ this.handleChange }
                            label={ `All ${ subject }s` }
                            id={ `${ scope }-radio-all` }
                            name={ group } />
                    </ListItem>
                    { selectable &&
                      <ListItem>
                          <Radio value={ SELECTED }
                              defaultChecked={ this.state.value === SELECTED }
                              onChange={ this.handleChange }
                              label={ `Only selected ${ subject }s` }
                              id={ `${ scope }-radio-selected` }
                              name={ group }  />
                      </ListItem> }
                    { selectable && this.state.value === SELECTED && <ListItem>{ children }</ListItem> }
                </BulletlessList>
            </React.Fragment>
        );
    }
}

export default RadioToggle;
