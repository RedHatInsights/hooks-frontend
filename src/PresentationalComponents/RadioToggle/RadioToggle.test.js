import { RadioToggle, ALL, SELECTED } from './RadioToggle';

describe('RadioToggle', () => {
    const scope = 'my-app-1';
    const selectable = true;
    const defaultProps = { scope, selectable };

    it('expect to render just the radios on all', () => {
        const wrapper = shallow(
            <RadioToggle { ...defaultProps } initial={ ALL }>
                Something
            </RadioToggle>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('expect to render children on selected', () => {
        const wrapper = shallow(
            <RadioToggle { ...defaultProps } initial={ SELECTED }>
                Something
            </RadioToggle>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('expect to render only one radio if not selectable', () => {
        const wrapper = shallow(
            <RadioToggle { ...defaultProps } initial={ SELECTED } selectable={ false }>
                Something
            </RadioToggle>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
