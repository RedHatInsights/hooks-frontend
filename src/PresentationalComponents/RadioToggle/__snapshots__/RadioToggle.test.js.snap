// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioToggle expect to render children on selected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioToggle
    initial="selected-only"
    scope="my-app-1"
    selectable={true}
    subject="event type"
  >
    Something
  </RadioToggle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <BulletlessList>
        <ListItem>
          <Radio
            className=""
            defaultChecked={false}
            id="my-app-1-radio-all"
            isChecked={null}
            isDisabled={false}
            isValid={true}
            label="All event types"
            name="my-app-1-event-type-radio"
            onChange={[Function]}
            style={
              Object {
                "marginBottom": ".5em",
                "marginTop": ".5em",
              }
            }
            value="all"
          />
        </ListItem>
        <ListItem>
          <Radio
            className=""
            defaultChecked={true}
            id="my-app-1-radio-selected"
            isChecked={null}
            isDisabled={false}
            isValid={true}
            label="Only selected event types"
            name="my-app-1-event-type-radio"
            onChange={[Function]}
            style={
              Object {
                "marginBottom": "1em",
                "marginTop": ".5em",
              }
            }
            value="selected-only"
          />
        </ListItem>
        <ListItem>
          Something
        </ListItem>
      </BulletlessList>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListItem>
            <Radio
              className=""
              defaultChecked={false}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />
          </ListItem>,
          <ListItem>
            <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-selected"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="Only selected event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                }
              }
              value="selected-only"
            />
          </ListItem>,
          <ListItem>
            Something
          </ListItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Radio
              className=""
              defaultChecked={false}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "defaultChecked": false,
              "id": "my-app-1-radio-all",
              "isChecked": null,
              "isDisabled": false,
              "isValid": true,
              "label": "All event types",
              "name": "my-app-1-event-type-radio",
              "onChange": [Function],
              "style": Object {
                "marginBottom": ".5em",
                "marginTop": ".5em",
              },
              "value": "all",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-selected"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="Only selected event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                }
              }
              value="selected-only"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "defaultChecked": true,
              "id": "my-app-1-radio-selected",
              "isChecked": null,
              "isDisabled": false,
              "isValid": true,
              "label": "Only selected event types",
              "name": "my-app-1-event-type-radio",
              "onChange": [Function],
              "style": Object {
                "marginBottom": "1em",
                "marginTop": ".5em",
              },
              "value": "selected-only",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Something",
          },
          "ref": null,
          "rendered": "Something",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BulletlessList>
          <ListItem>
            <Radio
              className=""
              defaultChecked={false}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />
          </ListItem>
          <ListItem>
            <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-selected"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="Only selected event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                }
              }
              value="selected-only"
            />
          </ListItem>
          <ListItem>
            Something
          </ListItem>
        </BulletlessList>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ListItem>
              <Radio
                className=""
                defaultChecked={false}
                id="my-app-1-radio-all"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="All event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": ".5em",
                    "marginTop": ".5em",
                  }
                }
                value="all"
              />
            </ListItem>,
            <ListItem>
              <Radio
                className=""
                defaultChecked={true}
                id="my-app-1-radio-selected"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="Only selected event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": "1em",
                    "marginTop": ".5em",
                  }
                }
                value="selected-only"
              />
            </ListItem>,
            <ListItem>
              Something
            </ListItem>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Radio
                className=""
                defaultChecked={false}
                id="my-app-1-radio-all"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="All event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": ".5em",
                    "marginTop": ".5em",
                  }
                }
                value="all"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "defaultChecked": false,
                "id": "my-app-1-radio-all",
                "isChecked": null,
                "isDisabled": false,
                "isValid": true,
                "label": "All event types",
                "name": "my-app-1-event-type-radio",
                "onChange": [Function],
                "style": Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                },
                "value": "all",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Radio
                className=""
                defaultChecked={true}
                id="my-app-1-radio-selected"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="Only selected event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": "1em",
                    "marginTop": ".5em",
                  }
                }
                value="selected-only"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "defaultChecked": true,
                "id": "my-app-1-radio-selected",
                "isChecked": null,
                "isDisabled": false,
                "isValid": true,
                "label": "Only selected event types",
                "name": "my-app-1-event-type-radio",
                "onChange": [Function],
                "style": Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                },
                "value": "selected-only",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Something",
            },
            "ref": null,
            "rendered": "Something",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`RadioToggle expect to render just the radios on all 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioToggle
    initial="all"
    scope="my-app-1"
    selectable={true}
    subject="event type"
  >
    Something
  </RadioToggle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <BulletlessList>
        <ListItem>
          <Radio
            className=""
            defaultChecked={true}
            id="my-app-1-radio-all"
            isChecked={null}
            isDisabled={false}
            isValid={true}
            label="All event types"
            name="my-app-1-event-type-radio"
            onChange={[Function]}
            style={
              Object {
                "marginBottom": ".5em",
                "marginTop": ".5em",
              }
            }
            value="all"
          />
        </ListItem>
        <ListItem>
          <Radio
            className=""
            defaultChecked={false}
            id="my-app-1-radio-selected"
            isChecked={null}
            isDisabled={false}
            isValid={true}
            label="Only selected event types"
            name="my-app-1-event-type-radio"
            onChange={[Function]}
            style={
              Object {
                "marginBottom": "1em",
                "marginTop": ".5em",
              }
            }
            value="selected-only"
          />
        </ListItem>
      </BulletlessList>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListItem>
            <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />
          </ListItem>,
          <ListItem>
            <Radio
              className=""
              defaultChecked={false}
              id="my-app-1-radio-selected"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="Only selected event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                }
              }
              value="selected-only"
            />
          </ListItem>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "defaultChecked": true,
              "id": "my-app-1-radio-all",
              "isChecked": null,
              "isDisabled": false,
              "isValid": true,
              "label": "All event types",
              "name": "my-app-1-event-type-radio",
              "onChange": [Function],
              "style": Object {
                "marginBottom": ".5em",
                "marginTop": ".5em",
              },
              "value": "all",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Radio
              className=""
              defaultChecked={false}
              id="my-app-1-radio-selected"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="Only selected event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                }
              }
              value="selected-only"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "defaultChecked": false,
              "id": "my-app-1-radio-selected",
              "isChecked": null,
              "isDisabled": false,
              "isValid": true,
              "label": "Only selected event types",
              "name": "my-app-1-event-type-radio",
              "onChange": [Function],
              "style": Object {
                "marginBottom": "1em",
                "marginTop": ".5em",
              },
              "value": "selected-only",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BulletlessList>
          <ListItem>
            <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />
          </ListItem>
          <ListItem>
            <Radio
              className=""
              defaultChecked={false}
              id="my-app-1-radio-selected"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="Only selected event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                }
              }
              value="selected-only"
            />
          </ListItem>
        </BulletlessList>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ListItem>
              <Radio
                className=""
                defaultChecked={true}
                id="my-app-1-radio-all"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="All event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": ".5em",
                    "marginTop": ".5em",
                  }
                }
                value="all"
              />
            </ListItem>,
            <ListItem>
              <Radio
                className=""
                defaultChecked={false}
                id="my-app-1-radio-selected"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="Only selected event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": "1em",
                    "marginTop": ".5em",
                  }
                }
                value="selected-only"
              />
            </ListItem>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Radio
                className=""
                defaultChecked={true}
                id="my-app-1-radio-all"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="All event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": ".5em",
                    "marginTop": ".5em",
                  }
                }
                value="all"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "defaultChecked": true,
                "id": "my-app-1-radio-all",
                "isChecked": null,
                "isDisabled": false,
                "isValid": true,
                "label": "All event types",
                "name": "my-app-1-event-type-radio",
                "onChange": [Function],
                "style": Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                },
                "value": "all",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Radio
                className=""
                defaultChecked={false}
                id="my-app-1-radio-selected"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="Only selected event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": "1em",
                    "marginTop": ".5em",
                  }
                }
                value="selected-only"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "defaultChecked": false,
                "id": "my-app-1-radio-selected",
                "isChecked": null,
                "isDisabled": false,
                "isValid": true,
                "label": "Only selected event types",
                "name": "my-app-1-event-type-radio",
                "onChange": [Function],
                "style": Object {
                  "marginBottom": "1em",
                  "marginTop": ".5em",
                },
                "value": "selected-only",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`RadioToggle expect to render only one radio if not selectable 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioToggle
    initial="selected-only"
    scope="my-app-1"
    selectable={false}
    subject="event type"
  >
    Something
  </RadioToggle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <BulletlessList>
        <ListItem>
          <Radio
            className=""
            defaultChecked={true}
            id="my-app-1-radio-all"
            isChecked={null}
            isDisabled={false}
            isValid={true}
            label="All event types"
            name="my-app-1-event-type-radio"
            onChange={[Function]}
            style={
              Object {
                "marginBottom": ".5em",
                "marginTop": ".5em",
              }
            }
            value="all"
          />
        </ListItem>
      </BulletlessList>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListItem>
            <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />
          </ListItem>,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "defaultChecked": true,
              "id": "my-app-1-radio-all",
              "isChecked": null,
              "isDisabled": false,
              "isValid": true,
              "label": "All event types",
              "name": "my-app-1-event-type-radio",
              "onChange": [Function],
              "style": Object {
                "marginBottom": ".5em",
                "marginTop": ".5em",
              },
              "value": "all",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        false,
        false,
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BulletlessList>
          <ListItem>
            <Radio
              className=""
              defaultChecked={true}
              id="my-app-1-radio-all"
              isChecked={null}
              isDisabled={false}
              isValid={true}
              label="All event types"
              name="my-app-1-event-type-radio"
              onChange={[Function]}
              style={
                Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                }
              }
              value="all"
            />
          </ListItem>
        </BulletlessList>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ListItem>
              <Radio
                className=""
                defaultChecked={true}
                id="my-app-1-radio-all"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="All event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": ".5em",
                    "marginTop": ".5em",
                  }
                }
                value="all"
              />
            </ListItem>,
            false,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Radio
                className=""
                defaultChecked={true}
                id="my-app-1-radio-all"
                isChecked={null}
                isDisabled={false}
                isValid={true}
                label="All event types"
                name="my-app-1-event-type-radio"
                onChange={[Function]}
                style={
                  Object {
                    "marginBottom": ".5em",
                    "marginTop": ".5em",
                  }
                }
                value="all"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "defaultChecked": true,
                "id": "my-app-1-radio-all",
                "isChecked": null,
                "isDisabled": false,
                "isValid": true,
                "label": "All event types",
                "name": "my-app-1-event-type-radio",
                "onChange": [Function],
                "style": Object {
                  "marginBottom": ".5em",
                  "marginTop": ".5em",
                },
                "value": "all",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          false,
          false,
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
